src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -103,6 +111,7 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
                         new ServerTelemetryChannel(),
                         provider.GetServices<ITelemetryInitializer>(),
                         provider.GetServices<ITelemetryModule>(),
+                        new ApplicationInsightsApplicationIdProvider(),
```

Comment:
So we don't want to use the same registered instance here?

Reply:
Yes, we have to initialize configuration for this host AND Active (static singleton) config. Active is initialized just once and lives till process dies. Function users access Active instance (as they don't have access to DI) if they want to manually track any telemetry from the function code.
ApplicaitonIdProvider is disposable and dies together with WebHost, but we don't want to same for the Active one.

So, I'd love to avoid using Active at all and never even initialize it, and it would be possible if there was a binding for TelemtryClient for functions. I think @brettsam mentioned that it is something he was considering. IF it's done we can get rid of Active config for good.

Reply:
Awesome! BTW, it would be a breaking change to stop initializing Active config, if we'll ever release current version. Active initialization is a new thing and has never been released yet.

If the binding is planned after v2 GA, I should probably remove  Active initialization so we don't break it later. What do you think?

Reply:
@lmolkova -- That may be a good idea to remove this initialization. Accessing `Active` makes me a little nervous b/c Functions can restart the host and dispose services at any time. Having some Static service that lives across that barrier seems troublesome.

I think it'd be much nicer if we prioritized the direct access to the registered service. And like you said, since we've never released this, it technically wouldn't be cutting anything.

Reply:
great, I'll send another PR to remove Active initialization